### PR TITLE
Validate cluster deployment delete request

### PR DESCRIPTION
**Pull request by zhujian7**
If there is any hypershift hosted cluster running on the cluster deployment, it is not allowed to delete

### CODE REVIEW

**Comment by zhujian7** [comment]
it is difficult to use e2e to test the webhook, so I added some integration tests, after this is merged, will submit a PR to modify the prow config.

**Comment by ldpliu** [comment]
it looks good to me

File: pkg/webhook/validating/webhook.go [hunk 1]
```diff
@@ -224,9 +229,43 @@ func (a *AdmissionHandler) validateManagedCluster(request *v1.AdmissionRequest)
...
+	for _, claim := range managedCluster.Status.ClusterClaims {
+		if claim.Name == constants.ClusterClaimHostedClusterCountZero && strings.EqualFold(claim.Value, "false") {
+			hasHostedClusters = true
+		}
+	}
+
+	// the ClusterClaimKeyHostedClusterCountZero is managed by the hypershift addon, so here we check if the addon
+	// is available;
+	// We only deny this delete request with 100% certainty, because this webhook on delete requests is dangerous
+	if hasHostedClusters && meta.IsStatusConditionPresentAndEqual(managedCluster.Status.Conditions,
```
Comment (qiujian16) [inline pkg/webhook/validating/webhook.go:262]: we could short circuit this with

```go
if !hasHostedClusters {
  returan a.responseAllowed()
}
```

**Review by qiujian16 — COMMENTED** [review]
/lgtm

looks good, just a minor comment

**Pull request merged**